### DISCUSSION TITLE
PdfInfo.Create method lacks meaningful exception

### DISCUSSION THREAD

**Discussion by akshay9070** [dlemstra/Magick.NET#1606] [Help]
I'm using PdfInfo.Create method to get count of pages for given pdf file. The project is deployed on server, which is handling multiple requests at same time. 

When same request is processed on local or other server its giving page count without any error.

All requests are throwing exception at PdfInfo.Create as  :

FailedToExecuteCommand "C:\Projects\gswin64c.exe" -q -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT --permit-file-read="C:/Users/Tomas/AppData/Local/Temp/temp.pdf" -sPDFPassword="qwert" -c "(C:/Users/Tomas/AppData/Local/Temp/temp.pdf) (r) file runpdfbegin pdfpagecount = quit"' (1) @ error/ghostscript-private.h/ExecuteGhostscriptCommand/74`

Note : Can reason for error be due to heavy cpu/memory usage on server, since its also running multiple services.
So can anyone please help why is error happening.

**Comment by dlemstra** [comment]
You will only get more information when Ghostscript is providing that information and it looks like that is not happening now. But I think I know what is causing this issue. Looking at the error message I noticed this filename: `C:/Users/Tomas/AppData/Local/Temp/temp.pdf`. Are multiple threads writing to that file at the same time? You should probably be using `Path.GetTempFileName()` or `Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString())` instead.

**Reply by akshay9070** [reply]
No file is already present at specificed location and no write operation is allowed.Thanks but can you explain how can Path.GetTempFileName() would solve issue.

**Reply by dlemstra** [reply]
When you have multiple threads and they all write to the `temp.pdf` file this means that sometimes you can have multiple threads accessing the same file? Using `Path.GetTempFileName()` would allow you to have a unique name for that temporary file every time.
